checks.d/gearmand.py

```diff
@@ -20,18 +23,36 @@ def _get_client(self,host,port):
         return gearman.GearmanAdminClient(["%s:%s" %
             (host, port)])
 
-    def _get_metrics(self, client, tags):
-        data = client.get_status()
+    def _get_metrics(self, client, specified_tasks, tags):
+        tasks = client.get_status()
         running = 0
         queued = 0
         workers = 0
 
-        for stat in data:
+        if len(specified_tasks) > MAX_NUM_TASKS:
+            raise Exception(
+                "The maximum number of tasks you can specify is {}.".format(MAX_NUM_TASKS))
+
+        if not len(specified_tasks) == 0:
+            tasks = [t for t in tasks if t['task'] in specified_tasks]
+
+        if len(tasks) > MAX_NUM_TASKS:
+            # Display a warning in the info page
+            self.warning(
+                "Too many tasks to fetch. You must choose the tasks you are interested in by editing the gearmand.yaml configuration file or get in touch with Datadog Support")
+
+        for stat in tasks[:MAX_NUM_TASKS]:
             running += stat['running']
             queued += stat['queued']
             workers += stat['workers']
 
-        unique_tasks = len(data)
+            task_tags = tags[:]
+            task_tags.append("task:{}".format(stat['task']))
+            self.gauge("gearman.running_by_task", running, tags=task_tags)
+            self.gauge("gearman.queued_by_task", queued, tags=task_tags)
+            self.gauge("gearman.workers_by_task", workers, tags=task_tags)
+
+        unique_tasks = len(tasks)
```

Comment:
I don't think the existing metrics should be subject to the limits we've set on the maximum number of tasks.

You could either:
- Move the submission of the existing metrics to before the new `*_by_task` metrics
- Submit the `*_by_task` metrics in a separate function that `_get_metrics` would call (and pass a copy of the `tasks` list to it so that it doesn't change the list for the existing metrics).